sdk/src/main/java/io/dapr/client/DaprClient.java

```diff
@@ -21,55 +22,82 @@
+   * @param event the event to be published.
+   * @param <T>   The type of event to be published, use byte[] for skipping serialization.
+   * @return a Mono plan of type Void.
    */
   <T> Mono<Void> publishEvent(String topic, T event);
 
   /**
    * Publish an event.
    *
-   * @param topic    the topic where the event will be published
-   * @param event    the event to be published
+   * @param topic    the topic where the event will be published.
+   * @param event    the event to be published.
    * @param metadata The metadata for the published event.
-   * @param <T>      The type of event to be published.
-   * @return a Mono plan of type Void
+   * @param <T>      The type of event to be published, use byte[] for skipping serialization.
+   * @return a Mono plan of type Void.
    */
   <T> Mono<Void> publishEvent(String topic, T event, Map<String, String> metadata);
 
   /**
-   * Invoke a service
+   * Invoke a service with all possible parameters, using serialization.
    *
    * @param verb    The Verb to be used for HTTP will be the HTTP Verb, for GRPC is just a metadata value.
-   * @param appId   The Application ID where the service is
+   * @param appId   The Application ID where the service is.
    * @param method  The actual Method to be call in the application.
-   * @param request The request to be sent to invoke the service
-   * @param clazz   the Type needed as return for the call
-   * @param <T>     the Type of the return
-   * @param <K>     The Type of the request.
-   * @return A Mono Plan of type clazz
+   * @param request The request to be sent to invoke the service.
+   * @param metadata Metadata (in GRPC) or headers (in HTTP) to be send in request.
+   * @param clazz   the Type needed as return for the call.
+   * @param <T>     the Type of the return, use byte[] to skip serialization.
+   * @param <K>     The Type of the request, use byte[] to skip serialization.
+   * @return A Mono Plan of type clazz.
+   */
+  <T, K> Mono<T> invokeService(Verb verb, String appId, String method, K request, Map<String, String> metadata, Class<T> clazz);
+
+  /**
+   * Invoke a service with all possible parameters, using serialization.
+   *
+   * @param verb    The Verb to be used for HTTP will be the HTTP Verb, for GRPC is just a metadata value.
+   * @param appId   The Application ID where the service is.
+   * @param method  The actual Method to be call in the application.
+   * @param request The request to be sent to invoke the service.
+   * @param clazz   the Type needed as return for the call.
+   * @param <T>     the Type of the return, use byte[] to skip serialization.
+   * @param <K>     The Type of the request, use byte[] to skip serialization.
+   * @return A Mono Plan of type clazz.
    */
-  <T, K> Mono<T> invokeService(String verb, String appId, String method, K request, Class<T> clazz);
+  <T, K> Mono<T> invokeService(Verb verb, String appId, String method, K request, Class<T> clazz);
```

Comment:
A metadata would be needed for the api call (metadata could be a map of string, string)

Reply:
The idea was to have variations where the metadata is optional. If we do all the permutations of request/response types, adding more to this would pollute our APIs with too many combinations (IMO). I can make metadata present in all of them, then.

Reply:
Will keep metadata in all and permute on Input/Output instead.